lib/elixir/lib/keyword.ex

```diff
@@ -1316,4 +1316,87 @@ defmodule Keyword do
   def size(keywords) do
     length(keywords)
   end
+
+  @doc """
+  Filters `keywords` i.e. returns a keyword list containing
+  only the elements from `keywords` for which the function
+  `fun` returns a truthy value.
+
+  See also `reject/2` which discards all elements where the function
+  returns a truthy value.
+
+  ## Examples
+
+      iex> Keyword.filter([one: 1, two: 2, three: 3], fn {_key, val} -> rem(val, 2) == 1 end)
+      [one: 1, three: 3]
+
+  """
+  @doc since: "1.13.0"
+  @spec filter(t, ({key, value} -> as_boolean(term))) :: t
+  def filter(keywords, fun) when is_list(keywords) and is_function(fun, 1) do
+    do_filter(keywords, fun)
+  end
+
+  defp do_filter([], _fun), do: []
+
+  defp do_filter([element | vals], fun) do
+    if fun.(element) do
+      [element | do_filter(vals, fun)]
+    else
+      do_filter(vals, fun)
+    end
+  end
+
+  @doc """
+  Returns a keyword list excluding the elements from `keywords`
+  for which the function `fun` returns a truthy value.
```

Comment:
```suggestion
  Returns a keyword list excluding the elements for which
  `fun` returns a truthy value.
```

Reply:
I wonder if we maybe should use the wording `entries` (rather than `elements` or `pairs`) since this is used in the rest of the `Keyword` module's documentation. What do you think?

Reply:
Agree. I didn't like elements and suggested pairs, but entries sounds better